tests/test-include-ostree-h.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2018 Endless Mobile, Inc.
+ *
+ * SPDX-License-Identifier: LGPL-2.0+
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ * Authors:
+ *  - Matthew Leeds <matthew.leeds@endlessm.com>
+ */
+
+#include "config.h"
+
+#include <glib.h>
+#include <locale.h>
+
+#if GLIB_CHECK_VERSION(2, 44, 0)
+#include <ostree.h>
+#endif
+
+static void
+test_include_ostree_h_compiled (void)
+{
+#if !GLIB_CHECK_VERSION(2, 44, 0)
+      g_test_skip ("Skipping test without glib >= 2.44 to avoid depending on"
+                   "libglnx for g_autoptr backports");
```

Comment:
Can you expand on this? Now that you've removed the libglnx include from public headers, this should work even on clients with glib < 2.44, right? Our `GLIB_VERSION_MIN_REQUIRED` is 2.40.

Reply:
`ostree.h` includes `ostree-autocleanups.h` which uses `G_DEFINE_AUTOPTR_CLEANUP_FUNC` which isn't defined on clients with glib < 2.44. It's different here than for a truly external program because `OSTREE_COMPILATION` is defined; see [here](https://github.com/ostreedev/ostree/blob/master/src/libostree/ostree-autocleanups.h#L34).

Reply:
Right, but if we want to really test this like an external program would, then we shouldn't have `-DOSTREE_COMPILATION` set, right? Offhand, it seems like adding `-UOSTREE_COMPILATION` should do the trick.

Reply:
You're right, that seems like a better solution. I thought it might have side effects but it looks like that symbol is only used in one place.